ast/parser.go

```diff
@@ -2306,6 +2306,11 @@ func (b *metadataParser) Parse() (*Annotations, error) {
 				b.loc = comment.Location
 			}
 		}
+
+		if match == nil && len(b.comments) > 0 {
```

Comment:
A small comment would be nice, like "use first comment's location for yaml errors if ..."

Also, can't we put it inside the block above, since it wouldn't matter if `err == nil`?

Reply:
Thanks! Fixing.

Reply:
Or wait, you're saying we should always set the location of `b` to the location of the first comment, and not just when there's an error?

Reply:
No I think what we have now is good. Thank you!